papermill/api.py

```diff
@@ -221,24 +221,54 @@ def display_output(self, name):
         if name not in outputs:
             raise PapermillException(
                 "Output Name '%s' is not available in this notebook.")
-        output = outputs[name]
-        ip_display(output.data, metadata=output.metadata, raw=True)
+
+        # Loop over elements of output
+        for output in outputs[name]:
+            ip_display(output.data, metadata=output.metadata, raw=True)
 
 
 def _get_papermill_metadata(nb, name, default=None):
     """Returns a dictionary of a notebook's metadata."""
     return nb.metadata.get('papermill', {}).get(name, default)
 
 
+def _cell_has_output_tag(cell):
+
+    for output in cell.get('outputs', []):
+        if 'papermill' in output.get('metadata', {}) and output.metadata.papermill.get('name'):
+            return True
+
+    return False
+
+
+def _get_cell_output_tag(cell):
+
+    if _cell_has_output_tag(cell):
+
+        names = set([output.metadata.papermill.get('name') for output in cell.outputs if 'papermill' in output.get('metadata', [])])
+
+        # If there are conflicting names, throw a shoe
+        if len(names) != 1:
+            raise NotImplemented
```

Comment:
I'd raise a `ValueError` and give a description for why it can't process names

Reply:
Is having more than one name in cell outputs not ok? Is that specified anywhere? We specifically have designed the notebook to be able to have multiple cell outputs, so this seems like an odd requirement.

Reply:
I'd prefer for the uniqueness criterion to be implemented one level up (in `_get_notebook_outputs`) rather than in this check. Here you're only getting the names themselves, the logic works whether you return one or many outputs. 

Importantly, by approaching it in this way we gain some forward compatibility. Even if we don't support multiple named outputs in a single cell today, if we were to begin to support that in the future, it won't require changing this API in order to enable that if it already returns an iterable. We'll then need to relax the uniqueness criterion in exactly one place (`_get_notebook_outputs`).